woocommerce/woocommerce-gutenberg-products-block
# Add the Interactivity API runtime to WooCommerce blocks

## What?

Includes the current [WP directives runtime](https://github.com/WordPress/block-hydration-experiments/issues/80) inside the WooCommerce blocks repository and makes it ready to use.

Tracking issue: https://github.com/woocommerce/woocommerce/issues/42486

## Why?

This is required to start testing the Interactivity API in WooCommerce blocks until it is natively available in WordPress.

## How?

- Copy-pasting the WP directives runtime code.
- Updating the Webpack configuration to compile the directive's JSX code into Preact.
- Adding an experimental filter to enable directives hydration (`false` by default).

## Testing instructions

See https://github.com/woocommerce/woocommerce-blocks/pull/8026#issuecomment-1382153094
